src/components/Visualization/StartScreen.js

```diff
@@ -11,26 +14,66 @@ import { sGetLoadError } from '../../reducers/loader.js'
 import { sGetUsername } from '../../reducers/user.js'
 import styles from './styles/StartScreen.module.css'
 
-const mostViewedVisualizationsQuery = {
-    mostViewedVisualizations: {
+const mostViewedQuery = {
+    mostViewed: {
         resource: 'dataStatistics/favorites',
-        params: ({ pageSize, username }) => ({
+        params: ({ username }) => ({
             eventType: EVENT_TYPE,
-            pageSize: pageSize || 10,
+            pageSize: 6,
             ...(username ? { username } : {}),
         }),
     },
 }
 
-const StartScreen = ({ error, username }) => {
-    const { data } = useDataQuery(mostViewedVisualizationsQuery, {
-        variables: { pageSize: 6, username },
+const visualizationsQuery = {
+    visualizations: {
+        resource: 'eventVisualizations',
+        params: ({ ids }) => ({
+            filter: `id:in:[${ids.join(',')}]`,
+            fields: ['id', 'displayName~rename(name)', 'type'],
+        }),
+    },
+}
+
+const useMostViewedVisualizations = (username) => {
+    const visualizations = useDataQuery(visualizationsQuery, {
+        lazy: true,
     })
 
-    const getContent = () =>
-        error ? (
-            getErrorContent()
-        ) : (
+    const mostViewed = useDataQuery(mostViewedQuery, {
+        variables: { username },
+        onComplete: (data) => {
+            visualizations.refetch({
+                ids: data.mostViewed.map((obj) => obj.id),
+            })
+        },
+    })
+
+    return {
+        mostViewed: visualizations.data
+            ? visualizations.data.visualizations.eventVisualizations
+            : undefined,
+        loading: mostViewed.loading || visualizations.loading,
+        fetching: mostViewed.fetching || visualizations.fetching,
+        error: mostViewed.error || visualizations.error,
+    }
+}
+
+const StartScreen = ({ error, username, setLoadError }) => {
+    const getContent = () => {
+        const data = useMostViewedVisualizations(username)
+
+        if (data.error) {
+            setLoadError(data.error)
+            return null
+        }
```

Comment:
I think this is fine, but you could also consider handling this from the `onError` callbacks of the `useDataQuery` hook options. I think it'd be effectively the same, but might look a bit more conventional than first calling an action and then rendering content.